### ISSUE TITLE
Suggestion:  Allowing the host to add people with a specific color

### ISSUE THREAD

**Issue by tronage** [pedrofracassi/amongcord#20]
Thank you so much for your hard work and efforts toward this, this is such a great tool that I think could use a tiny tweek.

As a streamer, playing this game with viewers can get a little hectic.
The concept is that *I* need to be in full control of the bot at all times so trolls/griefers don't run havoc.
I already see there is a suggestion to have a lock-down command so only the hose can control the bot, which is great!  In the mean-time, I made a work around so only "I" could send the bot commands, I created a special admin-amongcord channel that only I and the bot have access too (and removed the bot's access to general text channels).  Since nobody else can join, nobody else can see it, send it commands, or control it...

However, that removes the availability for them to use the ,join <color> command to add themselves to the bots player list...
I currently can use the `,joinall` and that will populate the list, but the colors are "random" (as it says it does).
This means I have to now mentally  ignore the colors (while using the phone sync feature) and do a mental what is their name on discord vs what is their name in game, to moderate their alive/dead status (or force them to use discord name as game name)...

My suggestion would be to have a command, something like `,forcejoin <@discordname> <color>`
This way I can personally populate the list with the appropriate and correct colors for the names. for examp in my hidden admin channel I could do a command:
`,forcejoin @someplayer red`
and that would populate the current game that the bot is tracking with that discord name as the color red.
OR an even more fancy version would be something like a  `,joinallnocolor` command that grabs all the names, and then spits back out the list (like the `,players` command does) and then you could just react to each name to set the color.

The react-to-set-color would be much better/easier but I'm sure would take more work.  I've noticed in my lobbies people have a habit of changing their color/skin between games, so before starting, you could just do a `,players` command, react to whom ever needs the appropriate color change to fix the colors.  Otherwise you probably would need a `,changecolor <@username> <new color>` command which would assign a player a new color (and if their new color was previously used, the bot would unassign that color from whom ever had it last, and would echo a warning about a player not having a color, and that they need to be assigned a new color.

I see this type of feature being great for everyone (not just a streamer) as it makes the setup easier since usually the host will be more skilled with using the bot, rather then explaining to people who may be new, how to join themselves into the game.  This makes it much easier to get the list populated with correct discord to game color relationships and lets you play the game more and mess around setting the bot up less!

Again thank you so much for this, and your efforts as it makes this game much more manageable.  Looking forward to a streamdeck plugin you mention may come in the future as that would be killer too!!!! Stay Safe and be well!

**Comment by bayoook** [comment]
up

**Comment by pedrofracassi** [comment]
First of all, than **you** for taking your time to write such a detailed issue. These really help.

> My suggestion would be to have a command, something like `,forcejoin <@discordname> <color>`

That's in the works! I'm sketching out a system that will let users with the right permission ("Mute Members", to be specific) target other users when running commands. There will be no need for new commands, as you'll just have to append the mention to the end of the command: `,join red @mention`, `,color blue @mention`, etc

> an even more fancy version would be something like a ,joinallnocolor command that grabs all the names, and then spits back out the list (like the ,players command does) and then you could just react to each name to set the color.

I don't think I see how you want react-to-join to work. Currently, I can only add reactions to the end of each message, not per line. So "react-to-join" would require me to send one message per player and then add the according reactions, _which takes time to do_, due to how Discord ratelimits the reaction endpoint.

Again, thank you so much for the detailed issue :)

**Comment by pedrofracassi** [comment]
Gonna go ahead and mention #7 just to link both issues and make it easier to track.

**Comment by tronage** [comment]
Awesomesauce!

> There will be no need for new commands, as you'll just have to append the mention to the end of the command: ,join red @mention, ,color blue @mention, etc

That works perfectly!  I didn't know you could use the same command with optional arguments! Fewer commands are better!  Can't wait for this to be released!!!

> I don't think I see how you want react-to-join to work.

Again not knowing the "backend" or how the programming works, I was thinking how with just general chat, I am able to type `Hello <enter> what's up <enter>` it will show on two lines under my name like this:

> Hello
> what's up

And I'm able to react to each line independently, so my thinking was that there could be a separate command from `,players` (I think that should stay as is) but maybe `,setplayercolors` that would only spit out just the names in separate lines.. For example:

> (input) 
> `,setplayercolors`
> (output)
> name1
> name2
> name3
> etc...

and I could right-click on a name, do a reaction on it, which would tell the bot what color to assign to that name.

In simpler terms, I guess what I'm suggesting is `,players` command wouldn't change (used as a status of who is in, and what colors they have).  While the `,setplayercolors` would just output a raw list of names, so they could be set as needed.  the workflow I envision would be something like:

> `,joinallnocolor`
> (bot grabs all the names in the voicechat, but no colors assigned)
> `,setplayercolors`
> (outputs the raw list of names is grabbed)
> right click and add reactions to each name to set the colors
> then just a final check to be sure all looks good you could do:
> `,players`
> (bot outputs the nice pretty list, as it currently does, to make sure all looks good before starting)
> `,sync`
> Fire up website on phone and start playing.

Again let me thank you for your hard work, and want you to know how useful this bot is for the Among Us community.!  Looking forward to the `,join <color> @mention` command!!!!

**Comment by pedrofracassi** [comment]
> Again not knowing the "backend" or how the programming works, I was thinking how with just general chat, I am able to type Hello <enter> what's up <enter> it will show on two lines under my name like this:
>> Hello
>> what's up

Yeah, that's two different messages, which probably wouldn't be possible because Discord limits how fast I can send stuff.

I like the idea of grabbing everyone in the voicechat without colors, though. Maybe I'll add some buttons on the app to let you set people's colors there.

**Comment by tronage** [comment]
> Maybe I'll add some buttons on the app to let you set people's colors there.

That would be awesome!  If you did implement that, would it apply to the website sync too, or just the android app (iOS user here)?  

Between the `,join <color> @mention` and the possibility of setting the colors on the phone, I'm super pumped about this right now!!!  Do you do paypal tips? I'm not into the Patreon thing but would love to buy you a coffee/beer to help you through a coding session or help contribute towards a streamdeck.

**Comment by pedrofracassi** [comment]
> If you did implement that, would it apply to the website sync too, or just the android app (iOS user here)?

Sync on the web and the Sync app share the same codebase. If one is updated, so is the other one! Same will happen to iOS once I release an app there.

**Comment by pedrofracassi** [comment]
> Do you do paypal tips?

Sure! Paypal email is the one on my GitHub profile.

**Comment by pedrofracassi** [comment]
`,forcejoin` is out! Ended up going with a new command to avoid making a mess with the systems I already had in place. Syntax is `,forcejoin <color> <@mention>`.
